test/unit/kubernetes-deploy/options_helper_test.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'tempfile'
+
+class OptionsHelperTest < KubernetesDeploy::TestCase
+  def test_single_template_dir_only
+    KubernetesDeploy::OptionsHelper.with_consolidated_template_dir([fixture_path('hello-cloud')]) do |template_dir|
+      assert_equal(fixture_path('hello-cloud'), template_dir)
+    end
+  end
+
+  def test_multiple_template_dirs
+    template_dirs = [fixture_path('hello-cloud'), fixture_path('partials')]
+
+    KubernetesDeploy::OptionsHelper.with_consolidated_template_dir(template_dirs) do |template_dir|
+      fixture_path_entries = template_dirs.collect { |dir| Dir.entries(dir) }.flatten.uniq
+      template_dir_entries = Dir.entries(template_dir)
+      assert_equal(fixture_path_entries.length, template_dir_entries.length)
+      fixture_path_entries.each do |fixture|
+        assert(template_dir_entries.select { |s| s.include?(fixture) })
+      end
+    end
+  end
+
+  def test_missing_template_dir_raises
+    assert_raises(KubernetesDeploy::OptionsHelper::OptionsError) do
+      KubernetesDeploy::OptionsHelper.with_consolidated_template_dir([]) do
+      end
+    end
+  end
+
+  def test_template_dir_with_stdin
+    old_stdin = $stdin
+
+    input = Tempfile.open("kubernetes_deploy_test")
+    File.open("#{fixture_path('for_unit_tests')}/service_test.yml", 'r') do |f|
```

Comment:
Same comment about using `File.join`.

Reply:
OTOH would it be simpler to use `StringIO` instead of actually manipulating files?

Reply:
Seeing as the tests are already written I'm inclined to leave them be.